src/ServiceInsight/ExtensionMethods/LayoutExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace ServiceInsight.ExtensionMethods
+{
+    using System;
+    using System.IO;
+    using Anotar.Serilog;
+    using DevExpress.Xpf.Bars;
+    using DevExpress.Xpf.Docking;
+    using DevExpress.Xpf.Grid;
+
+    public static class LayoutExtensions
+    {
+        public static string GetLayout(this BarManager barManager)
+        {
+            return TrySaveLayout(barManager);
+        }
+
+        public static string GetLayout(this DockLayoutManager dockLayoutManager)
+        {
+            return TrySaveLayout(dockLayoutManager);
+        }
+
+        public static string GetLayout(this GridControl gridControl)
+        {
+            return TrySaveLayout(gridControl);
+        }
+
+        public static void RestoreLayout(this BarManager barManager, Stream stream)
+        {
+            TryRestoreLayout(barManager, stream);
+        }
+
+        public static void RestoreLayout(this DockLayoutManager dockLayoutManager, Stream stream)
+        {
+            TryRestoreLayout(dockLayoutManager, stream);
+        }
+
+        public static void RestoreLayout(this GridControl gridControl, Stream stream)
+        {
+            TryRestoreLayout(gridControl, stream);
+        }
+
+        static string TrySaveLayout(dynamic control) //Lack of common interface :(
+        {
+            try
+            {
+                using (var ms = new MemoryStream())
+                {
+                    control.SaveLayoutToStream(ms);
+                    return ms.GetAsString();
+                }
+            }
+            catch (Exception ex)
+            {
+                LogTo.Information(ex, "Failed to save the layout, reason is: {ex}", ex);
```

Comment:
Is Info the right log level here? It doesn't cause the app to crash, but perhaps debug is appropriate.

Reply:
Given this is a non-critical failure, I'd say WARN is the appropriate level.

Reply:
After further review, I see that these log messages are pre-existing and have been moved from `Shell/Shell.xaml.cs`, i.e. the saving and restoration of the new settings for the column chooser are following the pattern of that of the existing settings for the layout of shell. For that reason I think we should keep this as-is for this PR. If we want to change the log level and messages (I think they also need some work) for problems in persisting all settings, I believe we should do that a a separate PR.

Reply:
👍  on doing it separately.